src/main/java/org/cryptomator/frontend/fuse/FileNameTranscoder.java

```diff
@@ -58,39 +65,42 @@ public ByteBuffer interpretAsFuseString(String fuseFileName) {
 
 	/**
 	 * Interprets the given string as NIO character set encoded and returns the original byte sequence.
+	 *
 	 * @param nioFileName string from the nio layer
 	 * @return A byte sequence with the original encoding of the input
 	 */
 	public ByteBuffer interpretAsNioString(String nioFileName) {
 		return nioCharset.encode(nioFileName);
 	}
 
-	private String transcode(String original, Charset srcCharset, Charset dstCharset, Optional<Normalizer.Form> srcNormalization, Optional<Normalizer.Form> dstNormalization) {
+	private String transcode(String original, Charset srcCharset, Charset dstCharset, Normalizer.Form dstNormalization, boolean applyNormalization) {
 		String result = original;
 		if (!srcCharset.equals(dstCharset)) {
 			result = dstCharset.decode(srcCharset.encode(result)).toString();
 		}
-		if (dstNormalization.isPresent()) {
-			if ((srcNormalization.isPresent() && !srcNormalization.get().equals(dstNormalization.get())) || srcNormalization.isEmpty()) {
-				Normalizer.normalize(result, dstNormalization.get());
-			}
+		if (applyNormalization) {
```

Comment:
Normalization should be skipped if
* either not required due to dst charset being non-unicode
* or src and dst normalization form are the same

Reply:
fixed in 2db9e87